lib/ansible/plugins/action/nxos.py

```diff
@@ -93,7 +93,10 @@ def run(self, tmp=None, task_vars=None):
                 provider['host'] = self._play_context.remote_addr
 
             if provider.get('port') is None:
-                provider['port'] = 80
+                if provider.get('use_ssl') is None or provider.get('use_ssl') is False:
```

Comment:
I was worried that `provider.get('use_ssl')` would generate a `NameError: not defined` if the `use_ssl` property is not configured but since the .get protects against this then I agree :). Will make the change.